pkg/filesystem/file.go

```diff
@@ -81,23 +79,11 @@ func CopyFile(from, to string) error {
 	return nil
 }
 
-// FileExistsAndNotDir checks if the file exists and its not a dir
-func FileExistsAndNotDir(filename string) bool {
-	info, err := os.Stat(filename)
-	if err != nil && os.IsNotExist(err) {
+// FileExistsAndNotDir checks if the file exists, and it's not a dir
+func FileExistsAndNotDir(path string, fs afero.Fs) bool {
+	info, err := fs.Stat(path)
+	if err != nil {
```

Comment:
Right now the only error that os.Stat can return is `os.NotExists`. I think it's safe to keep it like this but if they add another error other than that we can face some problems

Reply:
Then, why we discriminate in the other scenarios? I think both functions have to be consistent to what we want. Also, where did you see that it can only return that error?

Reply:
The reason why I've removed the check on `os.IsNotExist` is because in case `fs.Stat` fails, I don't think it's safe to let the execution continue and reach `info.IsDir` because `info` might be `nil` and locally it should not be checked.

At the same time in the other function `FileExistsWithFilesystem` I initially deleted the whole function, and then put it back because I am trying to do the smallest possible PR. 

This file and a lot of parts using this pkg need to be refactored so to unblock this PR now I will put back the check, we know future refactorings will come and we can consolidate the behavioir in the future.

Changed in ff715d15